openssl/src/ssl/mod.rs

```diff
@@ -1750,6 +1750,93 @@ impl SslContextBuilder {
+    ///
+    /// Requires OpenSSL 1.1.0 or newer.
+    #[corresponds(SSL_CTX_dane_enable)]
+    #[cfg(ossl110)]
+    pub fn dane_enable(&mut self) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::SSL_CTX_dane_enable(self.as_ptr())).map(|_| ()) }
+    }
+
+    /// Adjust the supported digest algorithms.
+    /// This must be done before any SSL handles are created for the context.
+    ///
+    /// The mtype argument specifies a DANE TLSA matching type and the md argument
+    /// specifies the associated digest algorithm handle.
+    /// The ord argument specifies a strength ordinal.
+    ///
+    /// Algorithms with a larger strength ordinal are considered more secure.
+    ///
+    /// Strength ordinals are used to implement RFC7671 digest algorithm agility.
+    ///
+    /// Specifying None for the digest algorithm for a matching type disables
+    /// support for that matching type.
+    ///
+    /// Matching type Full(0) cannot be modified or disabled.
+    ///
+    /// By default, matching type SHA2-256(1) (see RFC7218 for definitions of
+    /// the DANE TLSA parameter acronyms) is mapped to EVP_sha256() with a strength
+    /// ordinal of 1 and matching type SHA2-512(2) is mapped to EVP_sha512()
+    /// with a strength ordinal of 2.
+    ///
+    /// Requires OpenSSL 1.1.0 or newer.
+    #[corresponds(SSL_CTX_dane_mtype_set)]
+    #[cfg(ossl110)]
+    pub fn dane_mtype_set(
+        &mut self,
+        md: Option<crate::md::MdRef>,
+        mtype: DaneMatchType,
+        ord: u8,
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::SSL_CTX_dane_mtype_set(
+                self.as_ptr(),
+                md.map(|md| md.as_ptr()).unwrap_or(std::ptr::null_mut()),
+                mtype.as_raw(),
+                ord,
+            ))
+            .map(|_| ())
+        }
+    }
+
+    /// Disable server name checks when authenticating via DANE-EE(3) TLSA
+    /// records. For some applications, primarily web browsers, it is not safe to disable name
+    /// checks due to "unknown key share" attacks, in which a malicious server can convince a
+    /// client that a connection to a victim server is instead a secure connection to the malicious
+    /// server. The malicious server may then be able to violate cross-origin scripting
+    /// restrictions. Thus, despite the text of RFC7671, name checks are by default enabled for
+    /// DANE-EE(3) TLSA records, and can be disabled in applications where it is safe to do so. In
+    /// particular, SMTP and XMPP clients should set this option as SRV and MX records already make
+    /// it possible for a remote domain to redirect client connections to any server of its choice,
+    /// and in any case SMTP and XMPP clients do not execute scripts downloaded from remote
+    /// servers.
+    ///
+    /// Requires OpenSSL 1.1.0 or newer.
+    #[corresponds(SSL_CTX_dane_set_flags)]
+    #[cfg(ossl110)]
+    pub fn set_no_dane_ee_namechecks(&mut self) {
```

Comment:
We should just provide `dane_set_flags` etc methods rather than separate ones per flag.